src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.tsx

```diff
@@ -1,140 +1,117 @@
-        /** Whether or not a sign on form is loading (being submitted) */
-        isLoading: PropTypes.bool,
-
-        /** Whether or not the user has SAML enabled on their account */
-        isSAMLEnabled: PropTypes.bool,
-
-        /** Whether or not SAML is required on the account */
-        isSAMLRequired: PropTypes.bool,
-    }),
-
-    /** The credentials of the person signing in */
-    credentials: PropTypes.shape({
-        /** The login of the person signing in */
-        login: PropTypes.string,
-    }),
-
-    /** Session of currently logged in user */
-    session: PropTypes.shape({
-        /** Currently logged in user authToken */
-        authToken: PropTypes.string,
-    }),
+    account: OnyxEntry<Account>;
 
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
+    /** The credentials of the person logging in */
+    credentials: OnyxEntry<Credentials>;
 
-    /** Specifies autocomplete hints for the system, so it can provide autofill */
-    autoComplete: PropTypes.oneOf(['sms-otp', 'one-time-code']).isRequired,
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
 
-    /** Determines if user is switched to using recovery code instead of 2fa code */
-    isUsingRecoveryCode: PropTypes.bool.isRequired,
+type BaseValidateCodeFormProps = WithToggleVisibilityViewProps &
+    BaseValidateCodeFormOnyxProps & {
+        /** Specifies autocomplete hints for the system, so it can provide autofill */
+        autoComplete: 'sms-otp' | 'one-time-code';
 
-    /** Function to change `isUsingRecoveryCode` state when user toggles between 2fa code and recovery code */
-    setIsUsingRecoveryCode: PropTypes.func.isRequired,
+        /** Determines if user is switched to using recovery code instead of 2fa code */
+        isUsingRecoveryCode: boolean;
 
-    ...withLocalizePropTypes,
-};
+        /** Function to change `isUsingRecoveryCode` state when user toggles between 2fa code and recovery code */
+        setIsUsingRecoveryCode: (isUsingRecoveryCode: boolean) => void;
+    };
 
-const defaultProps = {
-    account: {},
-    credentials: {},
-    session: {
-        authToken: null,
-    },
-};
+type ValidateCodeFormVariant = 'validateCode' | 'twoFactorAuthCode' | 'recoveryCode';
 
-function BaseValidateCodeForm(props) {
-    const theme = useTheme();
+function BaseValidateCodeForm({account, credentials, session, autoComplete, isUsingRecoveryCode, setIsUsingRecoveryCode, isVisible}: BaseValidateCodeFormProps) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
+    const {translate} = useLocalize();
     const isFocused = useIsFocused();
-    const [formError, setFormError] = useState({});
-    const [validateCode, setValidateCode] = useState(props.credentials.validateCode || '');
+    const {isOffline} = useNetwork();
+    const [formError, setFormError] = useState<Partial<Record<ValidateCodeFormVariant, TranslationPaths>>>({});
```

Comment:
Let's put it into a type `Partial<Record<ValidateCodeFormVariant, TranslationPaths>>`